webpack/assets/javascripts/react_app/components/common/TypeAheadSelect/TypeAheadSelectSelectors.js

```diff
@@ -0,0 +1,15 @@
+import Immutable from 'seamless-immutable';
+
+export const mapSelected = selected => selected.map(item => item.label || item);
+
+const selectTypeAheadSelect = ({ typeAheadSelect }, id) =>
+  typeAheadSelect[id] || {};
+
+export const selectOptions = (state, id) => {
+  const typeAhead = selectTypeAheadSelect(state, id);
+  const options = typeAhead.options || [];
+  return Immutable.isImmutable(options) ? options.asMutable() : options;
```

Comment:
Why make it mutable?

Reply:
iirc making selections (of custom items?) was breaking because of something that the patternfly TypeAheadSelect is doing

Reply:
Makes sense.  It always makes me nervous when something so close to the Redux store is mutable, since you're not supposed to mutate anything in the store.  I'm assuming the Immutable `asMutable` method gives you a mutable copy, which is fine (rather than making the original mutable.)

Reply:
Exactly!